### PR TITLE
Full dist function

### PR DESCRIPTION
**Pull request by carlhiggs**
Pull request for fixes for point level estimates (incorrect in original SpatialVision code base, snapping to arbitrary node on network which can be hundreds of metres away --- ie. not true reflection of distance to closest, nor fair estimate of intersection or population density).  The changes in this branch relate to how NAs are treated, and there may be conflicts with output variable names --- but creating pull request now to make easier for input from others on this review.

### CODE REVIEW

File: process/sp.py [hunk 1]
```diff
@@ -23,6 +23,9 @@
 import json
 import fiona
 import sys
+# something about the CRS definition (see config file) elicits a future warning for Proj6; ignore for now
+import warnings
+warnings.simplefilter(action='ignore', category=FutureWarning)
```
Comment (gboeing) [inline process/sp.py]: CRS objects and definitions have changed in recent months. We should make sure these are up to date. Can you turn off the warning filtering? Better to see it.

File: process/sp.py [hunk 2]
```diff
@@ -34,7 +37,7 @@
 
     # the configuration file should put in the '/configuration' folder located at the same folder as scripts
     # load city-specific configeration file
-    jsonFile = 'configuration/'' + sys.argv[1]
+    jsonFile = 'configuration/' + sys.argv[1]
```
Comment (gboeing) [inline process/sp.py]: can this path building use `os.path.join` too?

**Review by gboeing — COMMENTED** [review]
Eyeballing it, this looks good. We don't have continuous integration tests so it's hard to guess visually how this will impact calculation and resulting data though...

File: process/sp.py (hunk 1 above)
Comment (carlhiggs) [reply process/sp.py]: Done!

File: process/sp.py (hunk 2 above)
Comment (carlhiggs) [reply process/sp.py]: Done!

**Review by gboeing — APPROVED** [review]

**Comment by gboeing** [comment]
@shiqin-liu do you have any feedback on this PR before we merge it into master?

**Comment by shiqin-liu** [comment]
> @shiqin-liu do you have any feedback on this PR before we merge it into master?

Thanks, Geoff. I think these look good to me, it is hard to think of the resulting changes by just reviewing this as you said. I think I will need to test the workflow on my end after this is merged into the master. 

I have a question though, what is `continuous integration tests`? I am not familiar with the term... could we establish one for the project?

**Comment by gboeing** [comment]
@shiqin-liu @carlhiggs ok, I will merge this into master today.

> what is continuous integration tests

These are tests that run every time a commit is made to master, or a PR is opened onto master. You can see how I handle CI/testing with [OSMnx here](https://github.com/gboeing/osmnx/tree/master/tests), at the bottom.

We should consider setting them up for this project. But it would require a wider discussion about what unit tests to include (and if we should code linting, docstring linting, etc).

**Comment by gboeing** [comment]
This has been merged. @shiqin-liu as time permits, could you run a quick test and make sure the workflow still works (ie, that output variable names don't conflict or anything like that)?

**Comment by shiqin-liu** [comment]
> This has been merged. @shiqin-liu as time permits, could you run a quick test and make sure the workflow still works (ie, that output variable names don't conflict or anything like that)?

Sure, will do!

**Pull request merged**